File: imob-erp-backend/src/main/java/com/imobcrm/lead/Lead.java
package com.imobcrm.lead;

import com.imobcrm.property.Property;
import com.imobcrm.shared.audit.Auditable;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "leads")
public class Lead extends Auditable {

    @Id
    private UUID id;

    @Column(name = "tenant_id", nullable = false)
    private UUID tenantId;

    @Column(name = "assigned_to", nullable = false)
    private UUID assignedTo;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String phone;

    private String email;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private LeadSource source;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private LeadStage stage;

    @Column(columnDefinition = "text")
    private String notes;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "lead_property",
            joinColumns = @JoinColumn(name = "lead_id"),
            inverseJoinColumns = @JoinColumn(name = "property_id"))
    @Builder.Default
    private Set<Property> propertiesOfInterest = new HashSet<>();
}

File: imob-erp-frontend/src/types/lead.ts
export type LeadSource = "WHATSAPP" | "SITE" | "INDICACAO" | "PORTAL_ZAP" | "PORTAL_VIVAREAL" | "OUTRO";
export type LeadStage = "NOVO" | "EM_ATENDIMENTO" | "VISITA_AGENDADA" | "PROPOSTA" | "FECHADO" | "PERDIDO";

export const LEAD_STAGES: LeadStage[] = [
  "NOVO",
  "EM_ATENDIMENTO",
  "VISITA_AGENDADA",
  "PROPOSTA",
  "FECHADO",
  "PERDIDO",
];

export interface Lead {
  id: string;
  name: string;
  phone: string;
  email?: string;
  source: LeadSource;
  stage: LeadStage;
  assignedTo: string;
  notes?: string;
  propertiesOfInterest: string[];
}

export type LeadRequest = Omit<Lead, "id" | "stage" | "propertiesOfInterest">;
